src/pages/address/[principal].tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import Head from 'next/head';
+import useSWR from 'swr';
+import {
+  AddressBalanceResponse,
+  MempoolTransaction,
+  MempoolTransactionListResponse,
+  TransactionResults,
+} from '@blockstack/stacks-blockchain-api-types';
+import { Box, Flex, Grid } from '@stacks/ui';
+import { Title } from '@components/typography';
+import type { NextPage, NextPageContext } from 'next';
+import { PageWrapper } from '@components/page';
+import { Section } from '@components/section';
+import { Rows } from '@components/rows';
+import { TransactionList } from '@components/transaction-list';
+import { microToStacks, truncateMiddle, validateStacksAddress } from '@common/utils';
+import { useApiServer } from '@common/hooks/use-api';
+import { getServerSideApiServer } from '@common/api/utils';
+import { fetchAllAccountData } from '@common/api/accounts';
+import { TokenBalancesRow } from '@components/balances/token-balances-row';
+import { Activity } from '@components/activity-row';
+import { StxBalances } from '@components/balances/stx-balance-card';
+import { getStackStartBlockHeight, hasTokenBalance } from '@common/utils/accounts';
+import { TokenBalancesCard } from '@components/balances/principal-token-balances';
+import { Meta } from '@components/meta-head';
+import { AddressNotFound } from '@components/address-not-found';
+
+const SummaryCard = ({ principal, hasTokenBalances, data }: any) => {
+  return (
+    <Section mb={'extra-loose'} title="Summary">
+      <Rows
+        px="base"
+        noTopBorder
+        items={[
+          {
+            label: {
+              children: 'Address',
+            },
+            children: principal,
+            copy: principal,
+          },
+          {
+            condition: !!data?.transactions?.results?.length,
+            label: {
+              children: 'Activity',
+            },
+            children: <Activity txs={data?.transactions?.results} />,
+          },
+          {
+            condition: !!hasTokenBalances,
+            label: {
+              children: 'Holdings',
+            },
+            children: <TokenBalancesRow balances={data?.balances} />,
+          },
+          {
+            label: {
+              children: 'Fees',
+            },
+            children: `${microToStacks(data?.balances?.stx?.total_fees_sent || 0)} STX`,
+          },
+        ]}
+      />
+    </Section>
+  );
+};
+
+interface AddressPageData {
+  principal: string;
+  balances?: AddressBalanceResponse;
+  transactions?: TransactionResults | MempoolTransactionListResponse;
+  pendingTransactions?: MempoolTransaction[];
+  error?: boolean;
+}
```

Comment:
Are there actually cases where the props for this component will be `undefined` - like `balances`, etc?

Because, it really complicates a bunch of the types on this page and related functions. If we can change the type to ensure it's not undefined (which, idk if is true!) then things could be cleaner.